### PR TITLE
feat: [CODE-2180]: Add draft boolean for github, azure and gitlab PRs

### PR DESCRIPTION
**Pull request by karansaraswat19**

### CODE REVIEW

File: scm/driver/azure/pr.go [hunk 1]
```diff
@@ -107,6 +107,7 @@ type pr struct {
 	SourceRefName         string      `json:"sourceRefName"`
 	TargetRefName         string      `json:"targetRefName"`
 	MergeStatus           string      `json:"mergeStatus"`
+	IsDraft               bool        `json:"isDraft"`
```
Comment (abhinav-harness) [inline scm/driver/azure/pr.go:110]: can we call it draft at all places and just rename json names?
I see we follow that convention - if not we can keep isDraft.

Comment (karansaraswat19) [reply scm/driver/azure/pr.go:110]: I read somewhere that the variable name need to start with uppercase letter if we expect it to be marshalled

**Comment by karansaraswat19** [comment]
> can you also check bitbucket cloud

> As of now, Bitbucket Cloud does not support draft pull requests. However, this feature is available in other major platforms as listed above.
They still don't support it

File: scm/driver/azure/pr.go (hunk 1 above)
Comment (abhinav-harness) [reply scm/driver/azure/pr.go:110]: I meant something - but nvm - we are good

**Review by abhinav-harness — APPROVED** [review]

**Pull request merged**